libfwupd/fwupd-common.c

```diff
@@ -245,7 +281,14 @@ fwupd_get_os_release_full(const gchar *filename, GError **error)
 GHashTable *
 fwupd_get_os_release(GError **error)
 {
+	g_autofree gchar *sw_vers = g_find_program_in_path("sw_vers");
 	g_return_val_if_fail(error == NULL || *error == NULL, NULL);
+
+	/* macOS */
+	if (sw_vers != NULL)
+		return fwupd_get_os_release_for_macos(sw_vers, error);
+
+	/* Linux and *BSD */
```

Comment:
I think we should be #ifdef for different OS here - there is no value to having a fallback for Linux do this.  It's just going to be an extra check on each OS that is unnecessary.

Reply:
TARGET_OS_MAC isn't super easy to use. Maybe like this instead? ^^